### PR TITLE
tests: added SAUL test application

### LINKED ISSUE #4817: saul: Missing tests
SAUL is currently not found anywhere in the tests directory. The only test we have today are the manual shell commands in examples/default

### PR DESCRIPTION
**Pull request by haukepetersen**
fixes #4817

### CODE REVIEW

File: tests/saul/main.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+int main(void)
+{
+    phydat_t res;
+    uint32_t last = xtimer_now();
+
+    puts("SAUL test application");
+
+    while (1) {
+        saul_reg_t *dev = saul_reg_get();
```
Comment (authmillenon) [inline tests/saul/main.c]: Not that I paid any attention to SAUL before-hand, but why is this a function and not just a global variable as e.g. `sched_active_pid`?

**Comment by gebart** [comment]
Good idea, though I was thinking that there needs to be some unit tests of the saul registry as well. (saul_reg_* functions)

File: tests/saul/main.c (hunk 1 above)
Comment (haukepetersen) [inline tests/saul/main.c]: because I don't like these :-)

No, good point, I will look into improving this.

Comment (authmillenon) [inline tests/saul/main.c]: Me neither, but proponents of this have the points that in most cases you can save a few bytes of ROM this way.

Comment (OlegHahm) [inline tests/saul/main.c]: I guess no one likes global variables, but getters are worse. ;)

Comment (haukepetersen) [inline tests/saul/main.c]: so how about #4907

**Comment by haukepetersen** [comment]
@gebart: I see. Are you okay if I look into that in a different PR?

**Comment by gebart** [comment]
That is OK, let's keep the issue "missing Saul tests" open as a reminder.

**Comment by haukepetersen** [comment]
sounds good!

**Comment by gebart** [comment]
needs rebase

**Comment by haukepetersen** [comment]
rebased

**Comment by cgundogan** [comment]
@gebart do you ACK this PR besides of the missing SAUL unittests?

**Comment by gebart** [comment]
Tested working with the LIS3DH SAUL sensor on Mulle. ACK
Optional improvement: write out the name of each sensor so that the values have more meaning to the user.

**Comment by authmillenon** [comment]
Could be done in a later PR I say. Let's merge.

**Pull request merged**